test/e2e/upgrade_test.go

```diff
@@ -344,6 +352,7 @@ var _ = Describe("BMO Upgrade", func() {
 
 	AfterEach(func() {
 		cleanup(ctx, upgradeClusterProxy, namespace, cancelWatches, e2eConfig.GetIntervals("default", "wait-namespace-deleted")...)
+		cleanup(ctx, upgradeClusterProxy, bmoIronicNamespace, bmoIronicCancelWatches, e2eConfig.GetIntervals("default", "wait-namespace-deleted")...)
```

Comment:
This will only delete the namespace and everything in it, but cluster scoped resources will stay. I think we should cleanup also the CRDs. How about we instead do something like `kubectl delete -k /path/to/upgrade/kustomization`? That should delete everything we added or modified when applying the kustomization so it should be pretty clean.

Reply:
Thanks. I thought of something similar to that, but CAPI test framework does not currently support deletion of the resources, so doing so would require a bit bigger changes. 
Do we have any cluster scoped resources other than CRDs? IMO keeping the CRDs should be acceptable.

Reply:
Hmm. Yes you're right. Now I realized there might be nasty issues that we fail to tests if CRDs are left there.
I'll come up with a solution based on `kubectl remove -k` wow.

Reply:
I couldn't find any better solution than using `exec` for this :sweat_smile: It was based on CAPI implementation for kubectlApply, so imo should be ok for us, though I don't particularly love it :sweat: